assets/js/src/newsletters/send.jsx

```diff
@@ -81,13 +78,14 @@ class NewsletterSend extends React.Component {
 
   isValid = () => jQuery('#mailpoet_newsletter').parsley().isValid();
 
-  isValidFromAddress = () => {
+  isValidFromAddress = async () => {
+    const addresses = await this.loadAuthorizedEmailAddresses();
     const fromAddress = this.state.item.sender_address;
     if (window.mailpoet_mta_method !== 'MailPoet') {
```

Comment:
Can you please put this condition as a very first thing in the function so that we avoid doing a useless API request in `loadAuthorizedEmailAddresses` ?